### PR TITLE
ft(registration-client-182201524):Users should be able Signup/Registe…

### PR DESCRIPTION
**Pull request by kalisaNkevin**
#### What does this PR do?

Users should be able Signup/Register to Barefoot Nomad

#### Description of Task to be completed?

`.`The sign up form should present to the user descriptive validation errors incase of any.
`.`The user should receive verification link on email upon successful registration

#### How should this be manually tested?

After cloning this repository, Switch to this branch:  `ft-registration-client-182201524`
The run: `yarn install `
To run the app:  `yarn start`
To run the test:  `yarn test`

#### Any background context you want to provide?

Fetching API from the backend and material UI framework for css 

#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/182201524

### CODE REVIEW

**Review by uwishema11 — APPROVED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by munyanezaarmel — COMMENTED** [review]
You don't have to import axios again it is already set you have to import API which found in utils and also backend UR should be in environment variables

**Review by ishimweeli — APPROVED** [review]

**Comment by kalisaNkevin** [comment]
> You don't have to import axios again it is already set you have to import API which found in utils and also backend UR should be in environment variables

**Review by munyanezaarmel — COMMENTED** [review]
Works fine but I can suggest when registration is successful a page should navigate to login screen instead of home screen also check coveralls

**Comment by ericus123** [comment]
Screenshot 2022-08-14 at 18 01 31

Fix the validations.
- I'm seeing errors for no reason.
- These errors are confusing.
`enter your username is equired`
either say `enter your username` or `username is equired`.
do same for other inputs as well

File: test/app.test.js [hunk 1]
```diff
@@ -2,7 +2,12 @@ import React from 'react'
...
 import App from '../src/components/App'
+import Signup from '../src/components/Authentications/RegisterUser'
 
 test('Initial render, Welcome to BareFoot Nomad frontEnd Tests', () => {
   render(<App />)
 })
+test('It should test all Sign up lines', () => {
+  render(<Signup />)
+})
+
```
Comment (ericus123) [inline test/app.test.js]: add a signup test file separately and add signup tests.
test the inputs individually as well.

File: src/Views/RegisterView.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import Signup from '../components/Authentications/RegisterUser'
+
+function RegisterView() {
+  return (
+    <div>
```
Comment (ericus123) [inline src/Views/RegisterView.js]: remove these div tags and have signup component only here

**Review by ericus123 — CHANGES_REQUESTED** [review]

File: test/registartion.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import Signup from '../src/components/Authentications/RegisterUser'
+
+test('It should test all Sign up lines', () => {
+  render(<Signup />)
+})
```
Comment (ericus123) [inline test/registartion.test.js]: is this the only test for registration ???

**Comment by ericus123** [comment]
Signup doesn't work???

**Comment by ericus123** [comment]
Make sure to use only material UI alerts

**Review by ericus123 — CHANGES_REQUESTED** [review]

**Comment by ericus123** [comment]
@kalisaNkevin  How is `dist` folder visible among changes?
Isn't it ignored??
Make sure it doesn't appear among git changes.

**Review by munyanezaarmel — APPROVED** [review]
add .dist in gitignore

**Pull request closed without merging**